File: frontend/src/models/booking.ts
export interface BookingRequest {
    eventId: number;
    ticketTypeId: number;
    userId: string;
    quantity: number;
}

export interface BookingResponse {
    id: string;
    eventId: number;
    ticketTypeId: number;
    userId: string;
    quantity: number;
    status: string;
    createdAt: string;
    expiresAt: string;
}

File: services/booking-service/src/main/java/com/ticket/system/booking/model/Booking.java
package com.ticket.system.booking.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "bookings")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long eventId; // Referencia al ID del Catálogo

    @Column(nullable = false)
    private Long ticketTypeId; // Referencia al tipo de entrada del Catálogo

    @Column(nullable = false)
    private String userId;

    @Column(nullable = false)
    private Integer quantity;

    @Enumerated(EnumType.STRING)
    private BookingStatus status;

    private LocalDateTime createdAt;
    private LocalDateTime expiresAt;
}